syntax = "proto3";

package server_services;

// Request message
message SendFileRequestBody {
  bytes file = 1;              // DTD file as bytes
  string file_mime = 2;        // MIME type of the file
  string file_name = 3;        // File name
}

// Response message
message SendFileResponseBody {
  bool success = 1;            // Indicates if the upload was successful
}

message SendFileChunksRequest {
  bytes data = 1;              // The chunk of the file
  string file_name = 2;        // File name (optional, sent in the first chunk or metadata)
}

message SendFileChunksResponse {
  bool success = 1;            // Indicates if the upload was successful
  string message = 2;          // Additional status message
}


message CsvToXmlRequest{
    string file_name = 1;
}

message CsvToXmlResponse{
    bool success = 1;
    string message = 2;
}


message FilterXmlRequest{
    string file_name = 1;
    string city_name = 2;
}

message FilterXmlResponse {
    bool success = 1;
    string xml_data = 2;  // XML serializado como string
    string message = 3;
}


// Service definition
service SendFileService {
  rpc SendFile(SendFileRequestBody) returns (SendFileResponseBody);
  rpc SendFileChunks(stream SendFileChunksRequest) returns (SendFileChunksResponse);
  rpc CsvToXml(CsvToXmlRequest) returns (CsvToXmlResponse);
  rpc FilterXml(FilterXmlRequest) returns (FilterXmlResponse);
}